package/MDAnalysis/analysis/msd.py

```diff
@@ -0,0 +1,213 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+Mean Squared Displacement --- :mod:`MDAnalysis.analysis.msd`
+==============================================================
+
+This module implements the calculation of Mean Squared Displacmements (MSD).
+MSDs can be used to characterise the speed at which particles move and has its roots
+in the study of Brownian motion.  For a thorough review see XXX et al. MSDs are computed from
+the following expression
+
+Where XX represents an ensemble average over 
+
+The computation of the MSD in this way can be computationally intensive due to it's N^2 scaling. 
+An algorithm to compute the MSD with Nlogn(N) scaling based on a Fast Fourier Transform is known and can be accessed by setting FFT=True.
+
+
+
+
+Classes and Functions
+---------------------
+
+.. autoclass:: MeanSquaredDisplacement
+
+
+"""
+
+from __future__ import division, absolute_import
+from six.moves import zip
+
+import os
+import errno
+import warnings
+import bz2
+import functools
+
+import numpy as np
+from scipy import fft,ifft
+import logging
+import MDAnalysis
+
+
+
+class MeanSquaredDisplacement(object):
```

Comment:
I know this is WIP, but just a quick comment. Any reason this isn't using AnalysisBase?

Reply:
Hey :)
Neither one of
-  _setup_frames
-  _single_frame

makes sense for an MSD as the value is defined in terms of lag-times over the whole trajectory rather than actual timesteps.

As _single_frames is compulsory when subclassing analysis-base I decided not to subclass. Any suggestions appreciated though.